### PR TITLE
Component: jquery.now deprecation

### LINKED ISSUE #2959: Deprecate jQuery.now
It's already just a reference to `Date.now`

### PR DESCRIPTION
**Pull request by ShashankaNataraj**
### Summary
#2959 jquery.now deprecation
### Checklist

Mark an `[x]` for completed items, if you're not sure leave them unchecked and we can assist.
- [x] All authors have signed the CLA at https://contribute.jquery.com/CLA/
- [ ] New tests have been added to show the fix or feature works
- [x] Grunt build and unit tests pass locally with these changes
- [x] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

Thanks! Bots and humans will be around shortly to check it out.
1. Replaced all occurances of jquery.now with Date.now
2. Moved the jquery.now function to deprecated.js

Fixes #2959

### CODE REVIEW

**Comment by ShashankaNataraj** [comment]
@mgol It looks like I screwed up by signing up on the jQuery CLA site again and now the commits wont go through and fail at the CLA check. resetting the name on my local repo and trying to push doesn't help anymore.

Can you please suggest what I can do to resolve this?

**Comment by mgol** [comment]
You need to squash all your commits into one and then edit the author info. Read https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History for more details about squashing; later click on Details near the failed CLA check and you'll see more info on the jQuery site.

**Comment by ShashankaNataraj** [comment]
@mgol Done. Thanks for your time and help!!

**Comment by dmethvin** [comment]
LGTM, the corresponding docs ticket is https://github.com/jquery/api.jquery.com/issues/947

**Comment by ShashankaNataraj** [comment]
@dmethvin Thank you! Will work on the other issue I said I'd work on 👍 

Im assuming no further action is required from my end for this PR and that this PR will be merged sometime in the future.

File: src/ajax/var/nonce.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 define( [
 	"../../core"
-], function( jQuery ) {
+], function() {
```
Comment (mgol) [inline src/ajax/var/nonce.js]: `core` shouldn't be required here, please remove the whole array.

File: src/core.js [hunk 2]
```diff
@@ -220,20 +220,6 @@ jQuery.extend( {
 		return obj != null && obj === obj.window;
 	},
 
-	isNumeric: function( obj ) {
```
Comment (mgol) [inline src/core.js]: This doesn't belong in this PR, it should only be about `jQuery.now` deprecation.

File: src/deprecated.js [hunk 3]
```diff
@@ -22,6 +22,9 @@ jQuery.fn.extend( {
 		return arguments.length === 1 ?
 			this.off( selector, "**" ) :
 			this.off( types, selector || "**", fn );
+	},
+	now: function() {
```
Comment (mgol) [inline src/deprecated.js]: Wouldn't `now: Date.now` be enough?

File: test/unit/core.js [hunk 4]
```diff
@@ -494,74 +494,6 @@ QUnit.test( "isFunction", function( assert ) {
 	} );
 } );
 
-QUnit.test( "isNumeric", function( assert ) {
```
Comment (mgol) [inline test/unit/core.js]: Wrong PR, you shouldn't touch `jQuery.isNumeric` in this one.

**Comment by mgol** [comment]
I added some comments. It seems this PR is mixed with the one deprecating `jQuery.isNumeric`, why?

**Comment by markelog** [comment]
@ShashankaNataraj are you still interested?

**Comment by ShashankaNataraj** [comment]
Hi @markelog yes, definitely, will try to fix whatever @mgol suggested in the coming days.

**Comment by mgol** [comment]
@ShashankaNataraj A friendly ping. :)

**Comment by timmywil** [comment]
@ShashankaNataraj One more ping for you. 😄

**Comment by ShashankaNataraj** [comment]
@mgol @timmywil Folks, got busy with some things and Im really in a crunch right now. I will close off this PR so someone else can take this up.
